### ISSUE TITLE
Ability to scroll back and forth while playing a song

### ISSUE THREAD

**Issue by kaelyavel** [bryanbraun/music-box-fun#23]

**Comment by bryanbraun** [comment]
Yeah, so an early version of the site actually supported this feature, but I ended up moving away from it. At the time, I was using IntersectionObserver instead of WebAudio to play notes, but it ended up making the tempo less consistent, especially in Safari browsers. I have [some notes in the code about it here](https://github.com/bryanbraun/music-box-fun/blob/70f1edc0ebd45ade75b5eacf27bd399125c752b6/site/public/js/subscribers/page-scroller.js#L13-L19).

But I think there's a couple options, while still keeping the consistency of Web Audio:

1. Lock scrolling during play (what we currently do)
2. If a song is playing and the user scrolls, immediately pause the song (kind of like we do when somebody changes the tempo mid-song).
3. If a song is playing and the user scrolls, immediately pause the song, until the scrolling has stopped, after which we automatically start playing again from the new position.

I think we'd have to experiment with option 2 or 3 to see if it feels like a more natural experience than the current approach.
